test/unit/unit_socket.c

```diff
@@ -39,13 +39,13 @@ START_TEST (test_socket)
     s = pico_socket_open(PICO_PROTO_IPV4, 99, NULL);
     fail_if(s != NULL, "Error got socket wrong parameters");
 
-    s = pico_socket_open(PICO_PROTO_IPV4, -109, NULL);
+    s = pico_socket_open(PICO_PROTO_IPV4, (uint16_t)-109, NULL);
```

Comment:
casting a negative number to `uint16_t`? Are you sure that's what's intended? (question applies a few times)